AdvLoggerPkg/Library/AdvancedLoggerAccessLib/AdvancedLoggerAccessLib.c

```diff
@@ -136,6 +168,40 @@ FormatPhasePrefix (
   return (UINT16)PhaseStringLen;
 }
 
+/**
+  FormatDebugLevelPrefix
+
+  Adds a debug level indicator to the message being returned.  If debug level is recognized and specified,
+  returns the debug_level prefix in from the AdvMsgEntryPrefix, otherwise raw debug level value is returned.
+
+  @param  MessageBuffer
+  @param  MessageBufferSize
+  @param  Phase
+
+  @retval Number of characters printed
+*/
+STATIC
+UINT16
+FormatDebugLevelPrefix (
+  IN CHAR8   *MessageBuffer,
+  IN UINTN   MessageBufferSize,
+  IN UINT32  DebugLevel
+  )
+{
+  UINTN  DebugLevelStringLen;
+
+  // Add the matched debug level
+  for (UINT8 i = 0; i < sizeof (debug_levels) / sizeof (debug_levels[0]); i++) {
+    if (DebugLevel == debug_levels[i].value) {

```

Comment:
DebugLevels are a bitwise map. 

Some debug messages in C code will use the bitwise map i.e. `DEBUG ((DEBUG_ERROR | DEBUG_INIT, "Message Here\n"));`

You will need to account for messages that may have multiple bits set.

```suggestion
    if ( (DebugLevel & debug_levels[i].value) == debug_levels[i].value) {

```

Reply:
Thanks Aaron, fixed